DREAM3DReviewFilters/GenerateFeatureIDsbyBoundingBoxes.cpp

```diff
@@ -0,0 +1,514 @@
+    return;
+  }
+
+  std::vector<size_t> cDims(1, 1);
+  m_BoxFeatureIdsPtr = getDataContainerArray()->getPrereqArrayFromPath<DataArray<int32_t>>(this, getBoxFeatureIDsArrayPath(), cDims);
+  if(nullptr != m_BoxFeatureIdsPtr.lock())
+  {
+    m_BoxFeatureIds = m_BoxFeatureIdsPtr.lock()->getPointer(0);
+  }
+
+  cDims[0] = 3;
+  m_BoxCenterPtr = getDataContainerArray()->getPrereqArrayFromPath<DataArray<float>>(this, getBoxCenterArrayPath(), cDims);
+  if(nullptr != m_BoxCenterPtr.lock())
+  {
+    m_BoxCenter = m_BoxCenterPtr.lock()->getPointer(0);
+  }
+
+  m_BoxDimsPtr = getDataContainerArray()->getPrereqArrayFromPath<DataArray<float>>(this, getBoxDimensionsArrayPath(), cDims);
+  if(nullptr != m_BoxDimsPtr.lock())
+  {
+    m_BoxDims = m_BoxDimsPtr.lock()->getPointer(0);
+  }
+
+  cDims[0] = 1;
+  m_FeatureIdsPtr =
+      getDataContainerArray()->createNonPrereqArrayFromPath<DataArray<int32_t>>(this, getFeatureIDsArrayPath(), 0, cDims); /* Assigns the shared_ptr<> to an instance variable that is a weak_ptr<> */
+  if(nullptr != m_FeatureIdsPtr.lock())
+  {
+    m_FeatureIds = m_FeatureIdsPtr.lock()->getPointer(0);
+  }
+
+  DataArrayPath path_bounding(getBoxFeatureIDsArrayPath().getDataContainerName(), getBoxFeatureIDsArrayPath().getAttributeMatrixName(), "");
+  AttributeMatrix::Pointer attrMat_bounding = getDataContainerArray()->getPrereqAttributeMatrixFromPath(this, path_bounding, -301);
+
+  if(getErrorCode() < 0)
+  {
+    return;
+  }
+
+  int32_t numFeatureIDs = attrMat_bounding->getNumberOfTuples();
+  std::vector<size_t> tDims(1, numFeatureIDs + 1);
+  m->createNonPrereqAttributeMatrix(this, getFeatureAttributeMatrixArrayPath(), tDims, m_DestAttributeMatrixType, AttributeMatrixID20);
+}
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+bool IsPointInBounds(float xmax, float xmin, float ymax, float ymin, float zmax, float zmin, float x, float y, float z)
+{
+  bool inBounds = false;
+
+  if((x < xmax) && (x > xmin) && (y < ymax) && (y > ymin) && (z < zmax) && (z > zmin))
+  {
+    inBounds = true;
+  }
+
+  return inBounds;
+}
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+void GenerateFeatureIDsbyBoundingBoxes::checkBoundingBoxImage()
+{
+  size_t totalNumFIDs = m_BoxFeatureIdsPtr.lock()->getNumberOfTuples();
+  size_t totalNumElementsDest = m_FeatureIdsPtr.lock()->getNumberOfTuples();
+  DataContainer::Pointer dc = getDataContainerArray()->getDataContainer(getFeatureIDsArrayPath().getDataContainerName());
+  ImageGeom::Pointer image = dc->getGeometryAs<ImageGeom>();
+  SizeVec3Type udims = image->getDimensions();
+  FloatVec3Type uorigin = image->getOrigin();
+  FloatVec3Type uspacing = image->getSpacing();
+
+  int64_t dims[3] = {
```

Comment:
```suggestion
  std::array<int64_t, 3> dims = {
```
Prefer std::array to C style arrays.